dev/build.py

```diff
@@ -0,0 +1,55 @@
+import argparse
+import contextlib
+import shutil
+import subprocess
+import sys
+from pathlib import Path
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="Build MLflow package.")
+    parser.add_argument(
+        "--package-type",
+        help="Package type to build. Default is 'dev'.",
+        choices=["skinny", "release", "dev"],
+        default="dev",
+    )
+    return parser.parse_args()
+
+
+@contextlib.contextmanager
+def restore_changes():
+    try:
+        yield
+    finally:
+        subprocess.check_call(["git", "restore", ":^dev/build.py"])
+
```

Comment:
Would it be helpful for diagnosing build-related issues to capture subprocess output and raise the full stack trace to stdout?

Something like:

```python
def run_subprocess(command):
    try:
        result = subprocess.run(command, capture_output=True, text=True, check=True)
        return result
    except subprocess.CalledProcessError as e:
        print(f"Command '{' '.join(command)}' failed with return code {e.returncode}")
        print(f"stdout:\n{e.stdout}")
        print(f"stderr:\n{e.stderr}")
        raise
``` 
?

Reply:
@BenWilson2 Thanks for the comment! `check_call` shows stdout and stderr by default.